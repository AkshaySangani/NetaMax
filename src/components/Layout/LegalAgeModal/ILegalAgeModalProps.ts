export interface ILegalAgeModalProps {
  /**
   * The isOpen to modal
   * @type {boolean}
   */
  isOpen: boolean;

  /**
   * The title to display
   * @type {string}
   */
  title: string;

  /**
   * The description to display
   * @type {string}
   */
  description: string;

  /**
   * The onClickModal to modal
   * @type {() => void}
   */
  onClickModal: (isUserUnderLegalAge: boolean) => void;
}
