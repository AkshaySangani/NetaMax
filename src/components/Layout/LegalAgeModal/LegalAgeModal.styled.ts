import styled from "styled-components";
import { responsive } from "styled/responsive";

import { Box, ModalContent } from "@chakra-ui/react";

export const ModalContentCustom = styled(ModalContent)`
  background: #ffffff;
  min-width: 100%;
  display: flex;
  flex-direction: column;
  justify-content: center;
  height: 100%;
  margin: 0;
  padding: 1rem;
  ${responsive.mdPlus} {
    background: transparent;
  }
`;

export const BodyContent = styled(Box)`
  width: 100%;
  height: 100%;
  background: linear-gradient(180deg, #c00200 0%, #f04d4b 111.71%);
  border-radius: 32px;
  position: relative;
  display: flex;
  flex-direction: column;
  align-items: center;
  padding: 0 1rem;
  ${responsive.mdPlus} {
    width: 55%;
    height: auto;
    padding: 0 6rem 2.5rem 6rem;
    justify-content: center;
  }
`;

export const BodyDescription = styled(Box)`
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  margin: auto 0;
  padding-bottom: 5rem;
  ${responsive.mdPlus} {
    padding-bottom: 0;
  }
`;

export const ArrowContent = styled(Box)`
  position: absolute;
  top: -6px;
  left: 50%;
  transform: translateX(-50%);
  width: 30px;
  height: 15px;
  background-color: #c00200;
  border-radius: 15px 15px 0 0;
  &:before {
    content: "";
    position: absolute;
    width: 0;
    height: 0;
    border: 12px solid transparent;
    border-bottom: 4px solid #c00200;
    transform: rotate(161deg);
    top: 4px;
    left: -10px;
  }
  &:after {
    content: "";
    position: absolute;
    width: 0;
    height: 0;
    border: 12px solid transparent;
    border-bottom: 4px solid #c00200;
    transform: rotate(-161deg);
    top: 4px;
    right: -10px;
  }
`;

export const ExclamationContent = styled(Box)`
  position: relative;
  width: 44px;
  height: 44px;
  border: 2px solid rgba(255, 255, 255, 0.4);
  border-radius: 50%;
  top: 0.5rem;
  display: flex;
  justify-content: center;
  align-items: center;
  margin-bottom: 1rem;
  &:after {
    content: "?";
    position: absolute;
    font-family: "Roboto";
    font-style: normal;
    font-weight: 700;
    font-size: 16px;
    line-height: 140%;
    color: #ffffff;
  }
`;
