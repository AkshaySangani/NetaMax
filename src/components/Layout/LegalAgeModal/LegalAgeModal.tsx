import React, { ReactElement } from "react";

import Boy from "styled/icons/LegalAge/Boy";
import NetaLogoFullColor from "styled/icons/NetaLogoFullColor";

import { Box, Modal, ModalBody, ModalFooter, ModalHeader, ModalOverlay } from "@chakra-ui/react";

import { ButtonComponent } from "./ButtonComponent/ButtonComponent";
import { ILegalAgeModalProps } from "./ILegalAgeModalProps";
import {
  ArrowContent,
  BodyContent,
  BodyDescription,
  ExclamationContent,
  ModalContentCustom,
} from "./LegalAgeModal.styled";

/**
 * The Modal component.
 * @param {ILegalAgeModalProps} props the component props.
 * @returns {ReactElement} the modal component.
 */
export const LegalAgeModal = ({
  isOpen,
  title,
  description,
  onClickModal,
}: ILegalAgeModalProps): ReactElement => {
  return (
    <Modal isOpen={isOpen} onClose={() => null}>
      <ModalOverlay backgroundColor="rgba(255, 255, 255, 0.9)" />
      <ModalContentCustom>
        <Box maxH="125px" display="flex" flex="1" marginBottom="1.5rem">
          <NetaLogoFullColor />
        </Box>
        <BodyContent>
          <ArrowContent />
          <ExclamationContent />
          <BodyDescription>
            <Box my={6}>
              <Boy />
            </Box>
            <ModalHeader
              color="#FFFFFF"
              fontWeight="900"
              lineHeight="40px"
              fontSize="30px"
              textAlign="center"
              py="0.5rem"
              px="0"
            >
              {title}
            </ModalHeader>
            <ModalBody
              color="#FFC1C1"
              fontSize="14px"
              lineHeight="20px"
              textAlign="center"
              py="0.5rem"
              px="0"
            >
              {description}
            </ModalBody>
            <ModalFooter>
              <Box mr="1rem">
                <ButtonComponent label="Si" onClick={() => onClickModal(false)} />
              </Box>
              <ButtonComponent label="No" onClick={() => onClickModal(true)} />
            </ModalFooter>
          </BodyDescription>
        </BodyContent>
      </ModalContentCustom>
    </Modal>
  );
};
