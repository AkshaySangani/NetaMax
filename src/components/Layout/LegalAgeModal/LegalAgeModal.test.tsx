import { render, fireEvent } from "@testing-library/react";

import { LegalAgeModal } from "./LegalAgeModal";

const modalPropsDefault = {
  isOpen: true,
  title: "Modal",
  description: "Description",
  onClickModal: () => null,
};

describe("Render ModalComponent component", () => {
  it("ModalComponent should be render correctly", () => {
    const { container } = render(<LegalAgeModal {...modalPropsDefault} />);
    expect(container).toMatchSnapshot();
  });

  it("ModalComponent should be not render, when isOpen is false", () => {
    const { queryByText } = render(<LegalAgeModal {...modalPropsDefault} isOpen={false} />);
    expect(queryByText("Modal")).not.toBeInTheDocument();
  });

  it("ModalComponent should on click, when click in button 'SI'", () => {
    const onClickMock = jest.fn();
    const { getByText } = render(
      <LegalAgeModal {...modalPropsDefault} onClickModal={onClickMock} />
    );
    fireEvent.click(getByText("Si"));
    expect(onClickMock).toBeCalledTimes(1);
  });

  it("ModalComponent should on click, when click in button 'NO'", () => {
    const onClickMock = jest.fn();
    const { getByText } = render(
      <LegalAgeModal {...modalPropsDefault} onClickModal={onClickMock} />
    );
    fireEvent.click(getByText("No"));
    expect(onClickMock).toBeCalledTimes(1);
  });
});
