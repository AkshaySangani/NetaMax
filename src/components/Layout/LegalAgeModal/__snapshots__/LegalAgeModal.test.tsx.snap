// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render ModalComponent component ModalComponent should be render correctly 1`] = `<div />`;
